components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/NewApiVersionMailNotifier.java

```diff
@@ -0,0 +1,150 @@
+                notificationDTO.setTitle((String) notificationDTO.getProperty(NotifierConstants.TITLE_KEY));
+                notificationDTO.setMessage((String) notificationDTO.getProperty(NotifierConstants.TEMPLATE_KEY));
+                notificationDTO = loadMailTemplate(notificationDTO);
+                message.setSubject(notificationDTO.getTitle());
+                message.setContent(notificationDTO.getMessage(), NotifierConstants.TEXT_TYPE);
+                message.setFrom(new InternetAddress(mailConfigurations.getFromUser()));
+
+                message.addRecipient(Message.RecipientType.TO, new InternetAddress(mail));
+
+                Transport.send(message);
+
+            } catch (MessagingException e) {
+                log.error("Exception Occurred during Email notification Sending", e);
+            }
+
+        }
+    }
+
+    /**
+     * Get UserNotifier Email Set.
+     */
+    private Set<String> getEmailNotifierList(NotificationDTO notificationDTO) throws
+            APIManagementException {
+        Set<String> mailSet = new HashSet<>();
+        Set<String> subscriberList = (Set<String>) notificationDTO.
+                getProperty(NotifierConstants.SUBSCRIBERS_PER_API);
+        if (!subscriberList.isEmpty()) {
+            for (Iterator<String> it = subscriberList.iterator(); it.hasNext(); ) {
+                String userName = it.next();
+                APIManagerFactory apiManagerFactory = APIManagerFactory.getInstance();
+                String userId = apiManagerFactory.getIdentityProvider().getIdOfUser(userName);
+                mailSet.add(apiManagerFactory.getIdentityProvider().getEmailOfUser(userId));
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Subscriber List is Empty");
+            }
+        }
+        return mailSet;
+    }
+
+    /**
+     * Class to Authenticate User.
+     */
+    private static class SMTPAuthenticator extends Authenticator {
+        public PasswordAuthentication getPasswordAuthentication() {
+            String username = SMTP_AUTH_USER;
+            String password = SMTP_AUTH_PWD;
+            return new PasswordAuthentication(username, password);
+        }
+    }
+
+    /**
+     * Retrieves the message configurations from notificationConfiguration and sets the notification properties to.
+     * NotificationDTO
+     */
+
+    public NotificationDTO loadMailTemplate(NotificationDTO notificationDTO) {
+        String title = notificationDTO.getTitle();
+        title = title.replaceAll("\\$1", (String) notificationDTO.getProperty(NotifierConstants.NEW_API_VERSION));
+        title = title.replaceAll("\\$2", (String) notificationDTO.getProperty(NotifierConstants.API_NAME));
+
+        String content = notificationDTO.getMessage();
+        if (content != null && !content.isEmpty()) {
+            content = content.replaceAll("\\$1", (String) notificationDTO.getProperty
```

Comment:
Can't we use better way than replacing https://stackoverflow.com/questions/3655424/string-replacement-in-java-similar-to-a-velocity-template.